pkg/operator/controller/controller.go

```diff
@@ -169,8 +169,10 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 				}
 			} else {
 				// Handle everything else.
+				if err := r.ensureClusterIngressFinalized(ingress); err != nil {
```

Comment:
Alternative for consideration: rename to `enforceFinalizer` and add to the `enforceEffectiveIngressDomain` and `enforceEffectiveHighAvailability` "invariant processing chain" above

Reply:
Oh wait, we need the other enforces _even when we handle deletes_, nevermind